### PR TITLE
Add triageparty-cli app

### PR DESCRIPTION
**Pull request by eddiezane**
Ref: https://github.com/kubernetes/k8s.io/issues/2460

### CODE REVIEW

**Review by spiffxp — APPROVED** [review]
/approve
I leave it to @ameukam to LGTM and remove the hold

**Comment by spiffxp** [comment]
I'm not sure what github account this corresponds to nor what new value goes in here
```
$ gcloud secrets versions list triage-party-github-token --project=kubernetes-public
NAME  STATE    CREATED              DESTROYED
1     enabled  2020-06-25T19:14:22  -
```

**Comment by ameukam** [comment]
> triage-

Created when I made the request for this token : https://github.com/kubernetes/org/issues/1903#issuecomment-650063797. 

We probably want to create a new token from k8s-infra-ci-robot.

**Comment by spiffxp** [comment]
IIRC api rate limits are applied per-account not per-token, so I'm not sure we want to create a new token off the k8s-infra-ci-robot account.  I think creating a new github user and handing the token off to us as was done for k8s-triage-party-sig-release-bot sounds fine.

**Comment by spiffxp** [comment]
From https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting (emphasis added)

>For API requests using Basic Authentication or OAuth, you can make up to 5,000 requests per hour. Authenticated requests are associated with the authenticated user, regardless of whether Basic Authentication or an OAuth token was used. **This means that all OAuth applications authorized by a user share the same quota of 5,000 requests per hour when they authenticate with different tokens owned by the same user.**

**Comment by ameukam** [comment]
> IIRC api rate limits are applied per-account not per-token, so I'm not sure we want to create a new token off the k8s-infra-ci-robot account. I think creating a new github user and handing the token off to us as was done for k8s-triage-party-sig-release-bot sounds fine.

Are we make this a general rule like _Any SIG interested to run a triage-party instance needs to bring his own Github Token_ or do we want to do a case by case ruling ?

**Comment by spiffxp** [comment]
>Are we make this a general rule like Any SIG interested to run a triage-party instance needs to bring his own Github Token or do we want to do a case by case ruling ?

I could go either way.  If these are accounts used by the project it would be bad if they were some random person's personal account or an account that we could lose control of.  On the other hand, centralizing the work of provisioning and holding credentials for up to 30 github accounts seems against our values.

Maybe the middle ground is that a kubernetes.io group should be used for the account's e-mail address?

**Comment by spiffxp** [comment]
FWIW, I am fine letting this merge and sorting who is responsible for placing a valid value in the referenced external secret after the fact, let me know how y'all want to proceed @eddiezane @ameukam

**Comment by eddiezane** [comment]
We could reuse the https://github.com/k8s-triage-party-sig-release-bot token or create another account?

How close are we with rate limiting for the current one? It should be in the logs.

**Comment by spiffxp** [comment]
... and we just refactored how external IP's are done aggh.  You'll probably need to rebase

I agree looking at the tokens left for the release triage party user sounds like a good idea, I don't have access to the account's credentials now so I'm not quite sure who owns it.

**Comment by spiffxp** [comment]
Again, I would be fine letting this merge and then taking care of getting a valid secret later.  I can probably pick this back up next week and push us forward if need back.

File: apps/triageparty-cli/secret.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+apiVersion: kubernetes-client.io/v1
+kind: ExternalSecret
+metadata:
+  name: github-token
```
Comment (ameukam) [inline apps/triageparty-cli/secret.yaml:4]:
```suggestion
  name: triage-party-github-token
```

File: apps/triageparty-cli/secret.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  namespace: triageparty-cli
+  labels:
+    app: triage-party
+spec:
+  backendType: gcpSecretsManager
+  projectId: kubernetes-public
+  data:
+  - key: triage-party-github-token
+    name: triage-party-github-token
```
Comment (ameukam) [inline apps/triageparty-cli/secret.yaml:13]:
```suggestion
    name: token
```

File: apps/triageparty-cli/service.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: triage-party-cli
+  namespace: triageparty-cli
+  labels:
+    app: triage-party
+spec:
+  type: NodePort
```
Comment (ameukam) [inline apps/triageparty-cli/service.yaml:9]:
```suggestion
  type: ClusterIP
```

File: apps/triageparty-cli/service.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: Service
+metadata:
```
Comment (ameukam) [inline apps/triageparty-cli/service.yaml:3]:
```suggestion
metadata:
  annotations:
    cloud.google.com/neg: '{"ingress": true}'
```

**Comment by eddiezane** [comment]
Rebased and made @ameukam's changes. I think I got the new external IP right but double check.

**Comment by ameukam** [comment]
/lgtm
@eddiezane Drop the hold when ready

**Comment by ameukam** [comment]
@eddiezane Global IP address created : 

```shell
gcloud compute addresses list --project kubernetes-public --filter="name:triage-party-cli-ingress-prod" --format="value(address)"
34.117.106.163
```

**Comment by eddiezane** [comment]
I believe I need someone with adequate permissions to run the deploy script.

```
eddiezane@cloudshell:~/k8s.io/apps/triageparty-cli (kubernetes-public)$ ./deploy.sh
configmap/triage-party-config-cli created
deployment.apps/triage-party created
ingress.networking.k8s.io/triage-party-cli created
service/triage-party-cli created
persistentvolumeclaim/triage-party-persistence created
Error from server (Forbidden): error when retrieving current configuration of:
Resource: "networking.gke.io/v1, Resource=managedcertificates", GroupVersionKind: "networking.gke.io/v1, Kind=ManagedCertificate"
Name: "cli-triage-k8s-io", Namespace: "triageparty-cli"
from server for: "certificate.yaml": managedcertificates.networking.gke.io "cli-triage-k8s-io" is forbidden: User "eddiezane@gmail.com" cannot get resource "managedcertificates" in API group "networking.gke.io" in the namespace "triageparty-cli": requires one of ["container.managedCertificates.get"] permission(s).
Error from server (Forbidden): error when retrieving current configuration of:
Resource: "kubernetes-client.io/v1, Resource=externalsecrets", GroupVersionKind: "kubernetes-client.io/v1, Kind=ExternalSecret"
Name: "triage-party-github-token", Namespace: "triageparty-cli"
from server for: "secret.yaml": externalsecrets.kubernetes-client.io "triage-party-github-token" is forbidden: User "eddiezane@gmail.com" cannot get resource "externalsecrets" in API group "kubernetes-client.io" in the namespace "triageparty-cli": requires one of ["container.thirdPartyObjects.get"] permission(s).
```

**Comment by ameukam** [comment]
I did the deployment:

```shell
ameukam@cloudshell:~/k8s.io/apps/triageparty-cli (kubernetes-public)$ kubectl -n triageparty-cli logs triage-party-7458d9b54b-qrkx6
W0923 04:29:25.944011       1 main.go:90] --config and CONFIG_PATH were empty, falling back to /app/config/config.yaml
I0923 04:29:25.944133       1 disk.go:60] cache dir is /app/triage-party/
I0923 04:29:25.944163       1 provider.go:58] loaded 40 byte GITHUB_TOKEN token from %!s(MISSING)
W0923 04:29:25.944174       1 provider.go:56] No token found in environment variable GITLAB_TOKEN (empty)
I0923 04:29:25.944182       1 main.go:129] triage runtime config: {Cache:/app/triage-party/ Repos:[] DebugNumbers:[] GitHubAPIURL: GitHubToken:b6ed6aa7141ff3996bd9a1b1d91e19c397015476 GitLabToken:}
I0923 04:29:25.944257       1 triage.go:150] 1938 bytes read from config
F0923 04:29:25.944344       1 main.go:136] load from /app/config/config.yaml: unmarshal: yaml: line 3: mapping values are not allowed in this context
```

Little error in the config. I opened https://github.com/kubernetes/k8s.io/pull/2785

**Pull request merged**